src/views/sources/addSourceModal.tsx

```diff
@@ -70,188 +105,216 @@ const AddSourceModal: React.FC<AddSourceModalProps> = ({
           console.error(err);
         }
       });
-  }, [getCredentials, sourceTypeValue]);
+  }, [typeValue, getCredentials]);
 
-  const onAdd = values => {
-    const payload = {
-      credentials: credentials.map(c => Number(c)),
-      hosts: values['hosts'].split(','),
-      name: values['name'],
-      port: values['port'] || (isNetwork ? '22' : '443'),
-      options: !isNetwork
-        ? {
-            ssl_cert_verify: sslProtocol !== 'Disable SSL' && sslVerify,
-            disable_ssl: sslProtocol === 'Disable SSL',
-            ...(sslProtocol !== 'Disable SSL' && { ssl_protocol: sslProtocol })
-          }
-        : {
-            use_paramiko: useParamiko
-          },
-      ...(!source && { source_type: sourceTypeValue }),
-      ...(source && { id: source.id })
-    };
-    onSubmit(payload);
+  const handleInputChange = useCallback(
+    (field: string, value: unknown) => {
+      setFormData({ ...formData, [field]: value });
+    },
+    [formData]
+  );
+
+  const filterFormData = useCallback(
+    (data = formData) => {
+      const { credentials, useParamiko, sslVerify, sslProtocol, name, hosts, port } = data;
+      return {
+        name: name,
+        credentials: credentials?.map(c => Number(c)),
+        hosts: hosts?.split(','),
+        port: port || (isNetwork ? '22' : '443'),
```

Comment:
Sorry for bringing it so late and in this venue, but the default ports are:

* [`22` for Network scan](https://github.com/quipucords/quipucords/blob/main/quipucords/api/source/serializer.py#L525)
* [`6443` for OpenShift](https://github.com/quipucords/quipucords/blob/main/quipucords/api/source/serializer.py#L537)
* [`443` for everything else](https://github.com/quipucords/quipucords/blob/main/quipucords/api/source/serializer.py#L539)

I think this logic needs to be a little more complex. And probably in some other places where we deal with ports, too.

Reply:
@ mirek If you are ok with this, I will add your observation to the docs and I will fix it in another PR. Because of the major changes we are doing here, it is "blocking" work in other PRs. I would like to get it merged as soon as possible.

Reply:
Sure, documenting and fixing later is fine! I think we all forgot that OpenShift uses special default port.

Reply:
added to the doc, will resolve conversation, ok?